os_is_not_exist_matcher.go

```diff
@@ -0,0 +1,32 @@
+package gomegamatchers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onsi/gomega/types"
+)
+
+func BeAnOsIsNotExistError() types.GomegaMatcher {
```

Comment:
There's still a constructor function for `BeFalse`: https://github.com/onsi/gomega/blob/3e93f011213c200aadafcb19ed56eb718381cfbd/matchers.go#L49-L51

Without the constructor you would have to express assertions as `Expect(true).To(BeFalseMatcher{})`.

It looks like the Gomega codebase puts all the constructors in [a separate file](https://github.com/onsi/gomega/blob/3e93f011213c200aadafcb19ed56eb718381cfbd/matchers.go), whereas this codebase has them in the same file as the matcher structs.